### PR TITLE
fix(Core/Player): Arena AFK

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9730#issuecomment-995604384

## Tests Performed:

- 
- 

## How to Test the Changes:

1.Read https://github.com/azerothcore/azerothcore-wotlk/issues/9730#issuecomment-995604384
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
I dont think this is correct

**Comment by Kitzunu** [comment]
If you look at wowpedia you find a lot of AFK stuff in BG's but nothing in Arenas. So I believe this is not blizzlike.

**Comment by Kitzunu** [comment]
do this in a module instead if it is desired

**Comment by laasker** [comment]
> If you look at wowpedia you find a lot of AFK stuff in BG's but nothing in Arenas. So I believe this is not blizzlike.

https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-general/136220-question-leave-arena-bg-macro.html (from 2008)
''Is there any sort of macro, that makes you leave the arena or the battleground with 1 click of it ?''
''/afk
how about that''

https://www.mmo-champion.com/threads/2207618-anyone-got-a-surrender-macro-arena-script
''Now if I got you correctly, you want to leave the match without having to click on the green eye icon and select Leave the match? If that's the case, you can type /afk in the match and it will kick you out.''

https://us.forums.blizzard.com/en/wow/t/leaving-arena/862254
''Is there a reason why typing in /away or /afk during an arena game to leave on a loss we aren’t rewarded with honor?''

i searched for some pvp videos but its almost impossible to find anyone typing /afk to leave since 99% of the videos people dont include loses, and you would leave the match if you or your partner died

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -1253,7 +1253,7 @@ void Player::ToggleAFK()
     ToggleFlag(PLAYER_FLAGS, PLAYER_FLAGS_AFK);
 
     // afk player not allowed in battleground
-    if (isAFK() && InBattleground() && !InArena())
+    if (isAFK() && InBattleground() && InArena())
```
Comment (UltraNix) [inline src/server/game/Entities/Player/Player.cpp:1256]: This breaks `/afk` in battlegrounds.

**Review by UltraNix — CHANGES_REQUESTED** [review]

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (IntelligentQuantum) [reply src/server/game/Entities/Player/Player.cpp:1256]: > This breaks `/afk` in battlegrounds.

Hmmm i didnt test this pr, i will check it out.

**Comment by laasker** [comment]
> > If you look at wowpedia you find a lot of AFK stuff in BG's but nothing in Arenas. So I believe this is not blizzlike.
> 
> i searched for some pvp videos but its almost impossible to find anyone typing /afk to leave since 99% of the videos people dont include loses, and you would leave the match if you or your partner died

was watching some VoD and found this in Classic TBC https://clips.twitch.tv/IgnorantFitGazelleDuDudu-1iQg6Aq5x-1anCpk
his partner dies, so to leave arena faster he just type /afk to leave the arena match

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -1253,7 +1253,7 @@ void Player::ToggleAFK()
     ToggleFlag(PLAYER_FLAGS, PLAYER_FLAGS_AFK);
 
     // afk player not allowed in battleground
-    if (isAFK() && InBattleground() && InArena())
+    if (isAFK() && (InBattleground() || InArena()))
```
Comment (UltraNix) [inline src/server/game/Entities/Player/Player.cpp:1256]: `InArena()` is not needed - `InBattleground()` means all battlegrounds, including arenas.

**Review by UltraNix — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum @UltraNix  what's the status of this PR?  does it need further testing/review? is it ready to be merged?

don't forget to update the workflow labels please

**Comment by UltraNix** [comment]
The code is okay to me. I'm not judging if it's blizz or not.

**Comment by IntelligentQuantum** [comment]
Need judg label xD

**Comment by Eronox** [comment]
Tested in Retail, both BGs and Arenas instantly kicks you out if you use `/afk `.

I'm pretty convinced it kicks you out for Vanilla as well.

in arena, this is said from https://us.battle.net/support/en/article/46427 without mentioning manual `/afk` at all

> Your character receives the idle debuff to let you know that you've been inactive for too long, or reported for being AFK. If you don't engage in PvP combat within 1 minute your character will receive the inactive debuff. You have 1 more minute to engage in PvP combat. If you don't, you will be removed from the battleground.

**Pull request merged**